### PR TITLE
doc: fixing ubuntu daily build recipes during feature freeze

### PR DESCRIPTION
**Pull request by blackboxsw**
We can't use new-upstream-snapshot to fix daily build recipes for cherry picks if we are in feature freeze because we don't want to pull in all upstream commits.

### CODE REVIEW

File: doc/ubuntu_release_process.md [hunk 1]
```diff
@@ -205,12 +197,35 @@ We have daily packaging recipes that upload to the [daily ppa](https://code.laun
...
+and builds the package from there.  Daily recipe failures occur under two conditions:
+ 1. Tip of master drifts enough that patches in the ubuntu/$release branch need to be refreshed/updated/dropped as upstream/master changes.
+ 1. You just finished a [cherry-pick an upstream commit](#cherry-pick-process) into an
+    ubuntu/$release branch and need to drop that cherry pick patch so daily package
+    recipes continue to build.
+
+#### Dropping a cherry-pick patch
+
+ * **During Ubuntu series feature freeze** we cannot use `new-upstream-snapshot` because it pulls in all upstream changes and we don't want to include other upstream commits if we need a second cherry pick during feature freeze:
+   1. LOCAL_PATCH_COMMIT=`git log --pretty=oneline debian/ | awk 'NR == 1{print $1}'`
```
Comment (OddBloke) [inline doc/ubuntu_release_process.md]: The formatting of these lines will be quite confusing; can we either replace this numbered list with a code block or ensure that the whole of each line is marked up appropriately?

File: doc/ubuntu_release_process.md [hunk 2]
```diff
@@ -205,12 +197,35 @@ We have daily packaging recipes that upload to the [daily ppa](https://code.laun
...
+The daily recipe for each release checks out tip of master, merges the ubuntu/$release
+and builds the package from there.  Daily recipe failures occur under two conditions:
+ 1. Tip of master drifts enough that patches in the ubuntu/$release branch need to be refreshed/updated/dropped as upstream/master changes.
+ 1. You just finished a [cherry-pick an upstream commit](#cherry-pick-process) into an
+    ubuntu/$release branch and need to drop that cherry pick patch so daily package
+    recipes continue to build.
+
+#### Dropping a cherry-pick patch
+
+ * **During Ubuntu series feature freeze** we cannot use `new-upstream-snapshot` because it pulls in all upstream changes and we don't want to include other upstream commits if we need a second cherry pick during feature freeze:
```
Comment (OddBloke) [inline doc/ubuntu_release_process.md]: Could you expand this with a bit more description of what the steps actually do? I can't follow along with them.

File: doc/ubuntu_release_process.md [hunk 3]
```diff
@@ -120,36 +120,28 @@ This is generally not the mechanism that is preferred for any release supported
...
+    $ git tag --delete xx-current >/dev/null 2>&1; git tag xx-current
+    ```
+1. Cherry pick UPSTREAM_COMMITISH
+   ```
+   $ git featch upstream
+   $ git checkout upstream/ubuntu/xenial -b ubuntu/xenial
+   $ cherry-pick dc2bd79949
+   ```
+1. From here you follow along with the snapshot upload process from [`dch --release` and beyond](#upstream-snapshot-process).
+1. Once published, fix daily builds by [reverting your cherry-pick'd patch](#when-the-daily-recipe-build-fails)
```
Comment (OddBloke) [inline doc/ubuntu_release_process.md]: Should these steps be in this section and not in "When the daily recipe build fails" below?

Comment (OddBloke) [reply doc/ubuntu_release_process.md]: Oh, no, this looks like the right place, but you appear to be dropping the `**Note**` that explains why we're doing this?

File: doc/ubuntu_release_process.md [hunk 4]
```diff
@@ -205,12 +197,35 @@ We have daily packaging recipes that upload to the [daily ppa](https://code.laun
 
 ### When the daily recipe build fails ###
 
-The daily recipe for each release checks out master and then merges the ubuntu/$release
-and builds the package from there.  From time to time, the patches in the
-ubuntu/$release branch need to be refreshed/updated as upstream/master changes.
+The daily recipe for each release checks out tip of master, merges the ubuntu/$release
+and builds the package from there.  Daily recipe failures occur under two conditions:
+ 1. Tip of master drifts enough that patches in the ubuntu/$release branch need to be refreshed/updated/dropped as upstream/master changes.
```
Comment (raharper) [inline doc/ubuntu_release_process.md]: s/drifts enough/has commits which modify code that the release branch patches touch.

File: doc/ubuntu_release_process.md [hunk 5]
```diff
@@ -205,12 +197,35 @@ We have daily packaging recipes that upload to the [daily ppa](https://code.laun
...
+ 1. You just finished a [cherry-pick an upstream commit](#cherry-pick-process) into an
+    ubuntu/$release branch and need to drop that cherry pick patch so daily package
+    recipes continue to build.
```
Comment (raharper) [inline doc/ubuntu_release_process.md]: I don't understand this section;  If I just cherry-picked an upstream commit; daily builds aren;t broken yet; they could be if the cherry-pick patch touches files that are currently patched in the release branch.  This sort of tells me that the ``cherry-pick`` tool probably should verify if there's a collision, no?

File: doc/ubuntu_release_process.md (hunk 1 above)
Comment (raharper) [reply doc/ubuntu_release_process.md]: Is this first command run immediately after running cherry-pick?  If so, let's put it all together in a single shell session so we can see the flow of the user-story here.

**Comment by blackboxsw** [comment]
Dropped in favor of #45

**Pull request closed without merging**